reflex/components/datadisplay/dataeditor.py

```diff
@@ -0,0 +1,350 @@
+
+
+# replace below with Literal
+# class GridRowMarkers(str, Enum):
+#     """Grid Row Markers."""
+
+#     NUMBER = "number"
+#     NONE = "none"
+#     CHECKBOX = "checkbox"
+#     BOTH = "both"
+#     CLICKABLE_NUMBER = "clickable-number"
+
+
+# @serializer
+# def serialize_gridcolumn_icon(icon: GridColumnIcons) -> str:
+#     """Serialize grid column icon.
+
+#     Args:
+#         icon: the Icon to serialize.
+
+#     Returns:
+#         The serialized value.
+#     """
+#     return "prefix" + str(icon)
+
+
+# class DataEditorColumn(Base):
+#     """Column."""
+
+#     title: str
+#     id: Optional[str] = None
+#     type_: str = "str"
+
+
+class DataEditor(NoSSRComponent):
+    """The DataEditor Component."""
+
+    tag = "DataEditor"
+    is_default = True
+    library: str = "@glideapps/glide-data-grid@^5.3.0"
+    lib_dependencies: List[str] = ["lodash", "marked", "react-responsive-carousel"]
+
+    # number of rows
+    rows: Var[int]
+
+    # headers of the columns for the data grid
+    columns: Var[List[Dict[str, Any]]]
+
+    # the data
+    data: Var[Any]
+
+    # the name of the callback used to find the data to display
+    get_cell_content: Var[str]
+
+    # allow copy paste or not
+    get_cell_for_selection: Var[bool]
+
+    on_paste: Var[bool]
+
+    # Controls the drawing of the focus ring.
+    draw_focus_ring: Var[bool]
+
+    # Enables or disables the overlay shadow when scrolling horizontally.
+    fixed_shadow_x: Var[bool]
+
+    # Enables or disables the overlay shadow when scrolling vertically.
+    fixed_shadow_y: Var[bool]
+
+    # The number of columns which should remain in place when scrolling horizontally. Doesn't include rowMarkers.
+    freeze_columns: Var[int]
+
+    # Controls the header of the group header row.
+    group_header_height: Var[int]
+
+    # Controls the height of the header row.
+    header_height: Var[int]
+
+    # Additional header icons: (TODO: must be a map of name: svg)
+    header_icons: Var[Any]
+
+    # The maximum width a column can be automatically sized to.
+    max_column_auto_width: Var[int]
+
+    # The maximum width a column can be resized to.
+    max_column_width: Var[int]
+
+    # The minimum width a column can be resized to.
+    min_column_width: Var[int]
+
+    # Determins the height of each row.
+    row_height: Var[int]
+
+    # kind of raw marker (use Literal API ?)
+    row_markers: Var[str]
+
+    # Changes the starting index for row markers.
+    row_marker_start_index: Var[int]
+
+    # Sets the width of row markers in pixels, if unset row markers will automatically size.
+    row_marker_width: Var[int]
+
+    # enable horizontal smooth scrolling
+    smooth_scroll_x: Var[bool]
+
+    # enable vertical smooth scrolling
+    smooth_scroll_y: Var[bool]
+
+    # Controls the drawing of the left hand vertical border of a column. If set to a boolean value it controls all borders.
+    # TODO: need to support a mapping (dict[int, bool] for handling separate columns)
```

Comment:
Can you check how this gets rendered w/ the website docs later? I think we may not support having multiline comments for props.

Reply:
I've changed how I wrote it, should be good now.